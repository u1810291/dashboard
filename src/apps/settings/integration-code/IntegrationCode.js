import React from 'react'
import { connect } from 'react-redux'
import { SyntaxHighlighter } from 'src/components/syntax-highlighter'
import { getIntegrationCode } from 'src/state/merchant'
import Support from 'src/fragments/account/support'
import MatiDocs from 'src/fragments/account/mati-docs'
import { FormattedMessage } from 'react-intl'
import SettingsLayout from '../SettingsLayout'

export default
@connect(
  ({ auth: { token }, merchant: { integrationCode } }) => ({
    token,
    integrationCode
  }),
  {
    getIntegrationCode
  }
)
class IntegrationCode extends React.Component {
  componentDidMount() {
    this.props.getIntegrationCode(this.props.token)
  }

  render() {
    return (
      <React.Fragment>
        <h1>
          <FormattedMessage id="settings.integrationCode.title" />
        </h1>

        <SettingsLayout>

          <main>
            <SyntaxHighlighter
              dark={false}
              language="html"
              copyToClipboard
              code={this.props.integrationCode}
            />
            <MatiDocs />
          </main>

          <aside>
            <Support />
          </aside>

        </SettingsLayout>
      </React.Fragment>
    )
  }
}
