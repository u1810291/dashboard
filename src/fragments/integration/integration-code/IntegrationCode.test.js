import React from 'react'
import { shallow } from 'enzyme'
import IntegrationCode from '.'

describe('IntegrationCode', () => {
  test('component render', () => {
    const component = shallow(<IntegrationCode integrationCode="hello" />)
    expect(component).toMatchSnapshot()
  })
})
