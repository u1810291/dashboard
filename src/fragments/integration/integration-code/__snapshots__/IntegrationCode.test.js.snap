// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntegrationCode component render 1`] = `
<PanelBody>
  <Sections>
    <h3>
      <FormattedMessage
        id="fragments.integration.integration-code.title"
        values={Object {}}
      />
    </h3>
    <Items
      smallGap={true}
    >
      <Button
        buttonStyle="primary"
        size="small"
      >
        <FormattedMessage
          id="fragments.integration.integration-code.javascript"
          values={Object {}}
        />
      </Button>
      <Button
        buttonStyle="default"
        disabled={true}
        size="small"
      >
        <FormattedMessage
          id="fragments.integration.integration-code.react"
          values={Object {}}
        />
      </Button>
      <Button
        buttonStyle="default"
        disabled={true}
        size="small"
      >
        <FormattedMessage
          id="fragments.integration.integration-code.angular"
          values={Object {}}
        />
      </Button>
    </Items>
    <SyntaxHighlighter
      code="hello"
      language="html"
    />
  </Sections>
</PanelBody>
`;
